transmogrifier/sources/datacite.py

```diff
@@ -249,23 +265,23 @@ def create_from_datacite_xml(
             kwargs["publication_information"] = [publisher.string]
 
         # related_items, uses related_identifiers retrieved for identifiers
-        for related_identifier in related_identifiers:
-            ri = timdex.RelatedItem(
-                uri=cls.generate_related_item_identifier_url(related_identifier)
+        for related_identifier in [i for i in related_identifiers if i.string]:
+            kwargs.setdefault("related_items", []).append(
+                timdex.RelatedItem(
+                    uri=cls.generate_related_item_identifier_url(related_identifier),
+                    relationship=related_identifier.get("relationType"),
+                )
             )
-            if "relationType" in related_identifier.attrs:
-                ri.relationship = related_identifier.attrs["relationType"]
-            kwargs.setdefault("related_items", []).append(ri)
 
         # rights
-        rights_list = xml.metadata.find_all("rights")
-        for rights in rights_list:
-            r = timdex.Rights()
-            if rights.string:
-                r.description = rights.string
-            if "rightsURI" in rights.attrs:
-                r.uri = rights.attrs["rightsURI"]
-            kwargs.setdefault("rights", []).append(r)
+        for rights in xml.metadata.find_all("rights"):
```

Comment:
```
for right in [
    r for r in xml.metadata.find_all("rights") if r.string or r.get("rightsURI")
]:
    kwargs.setdefault("rights", []).append(
        timdex.Rights(
            description=right.string or None, uri=right.get("rightsURI")
        )
    )
```

Reply:
Much appreciated, I wasn't able to see that